docs/reference_guides/model_libraries/generic/costing/process_costing.rst

```diff
@@ -0,0 +1,865 @@
+    results = opt.solve(m, tee=True)
+
+Units
+^^^^^
+
+It is important to highlight that the costing method interrogates the property 
+package to determine the units of this model, if the user provided the correct 
+units in the metadata dictionary (see property models for additional information), 
+the model units will be converted to the right units. 
+For example: in this example area is in m^2, while the cost correlations for heat 
+exchangers require units to be in ft^2. Therefore, the costing method will convert 
+the units to ft^2. The use of Pyomo-unit conversion tools is under development.
+
+Available IDAES Costing Module Methods
+--------------------------------------
+
+A default costing module has been developed primarily based on base cost and purchase cost correlations 
+from the following reference with some exceptions (noted in the documentation as appropiate).
+
+Process and Product Design Principles: Synthesis, Analysis, and Evaluation. Seider, Seader, Lewin, Windagdo, 3rd Ed. John Wiley and Sons. Chapter 22. Cost Accounting and Capital Cost Estimation
+
+Users should refer to the reference above for details of the costing correlations, however, a summary of this methods is provided below.
+    
+Table 2. Cost basis for each unit model.
+
+=========================== =========================  ===========
+Unit Model                  Basis                      Units      
+=========================== =========================  ===========
+heat exchanger              :math:`area`               ft^2       
+pump                        :math:`fluid_{work}`       ft^3/s     
+compressor                  :math:`mechanical_{work}`  hp         
+turbine                     :math:`mechanical_{work}`  hp         
+vessels                     :math:`D , L`              ft         
+fired heaters               :math:`heat\_duty`         BTU/hr         
+=========================== =========================  ===========
+
+Unit Mapping
+^^^^^^^^^^^^
+
+As a convenience for users, common unit models are mapped with appropriate costing methods. Therefore, for quick costing with defaults, users may import the `unit_mapping` dictionary and pass the unit model type as below:
+
+.. code:: python
+
+    unit_mapping = {  
+            CSTR: cost_vertical_vessel,  
+            Compressor: cost_compressor,  
+            Flash: cost_vertical_vessel,  
+            Heater: cost_fired_heater,  
+            HeatExchanger: cost_heat_exchanger,  
+            HeatExchangerNTU: cost_heat_exchanger,  
+            PFR: cost_horizontal_vessel,  
+            PressureChanger: cost_pressure_changer,  
+            Pump: cost_pump,  
+            StoichiometricReactor: cost_horizontal_vessel,  
+            Turbine: cost_turbine,  
+        }
+
+    costing_method = unit_mapping[type(blk)]
+
+The mapping is inheritance aware, e.g. pumps will try costing with `cost_pump()` and fall back to `cost_pressure_changer()`.
```

Comment:
I would word this is less computer science based terms (i.e. inheritance); note that their is a hierarchy and that the framework will find the best match. Maybe mention inheritance in parenthesis so that those more familiar with the term will understand, but those who do not don't get confused.